plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -984,7 +985,12 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta interface
 	if node == nil {
 		return false, nil, fmt.Errorf("node not found")
 	}
-	if !c.satisfiesExistingPodsAntiAffinity(pod, meta, node) {
+	allPods, err := c.podLister.List(labels.Everything())
```

Comment:
@ravisantoshgudimetla we should probably do that, but that change does not belong in this PR. It is an optimization of the predicate function we should be done separately IMO.